src/AspNet.Security.OAuth.LinkedIn/LinkedInAuthenticationHandler.cs

```diff
@@ -24,7 +24,12 @@ public LinkedInAuthenticationHandler([NotNull] HttpClient client)
 
         protected override async Task<AuthenticationTicket> CreateTicketAsync([NotNull] ClaimsIdentity identity,
             [NotNull] AuthenticationProperties properties, [NotNull] OAuthTokenResponse tokens) {
-            var request = new HttpRequestMessage(HttpMethod.Get, Options.UserInformationEndpoint);
+            var endpoint = string.Empty;
+            if (Options.Fields.Count > 0) {
+                endpoint = $"{Options.UserInformationEndpoint}/~:({string.Join(",", Options.Fields)})";
```

Comment:
Yeah, not sure what other alternatives there are. LinkedIn requires that URL to be in a specific format and from past experience with them, any slight deviation from what they require will cause errors as their API does not seem to be very resilient.

Only alternative is to specify a placeholder for the fields in the `UserInfoEndpoint`, e.g.

```
public const string UserInformationEndpoint = "https://api.linkedin.com/v1/people/~:({0})"
```

This way you will know exactly where to insert the fields, and it allows people to add other query string parameters if they want to. This will have to be documented clearly though...